homeassistant/components/light/iglo.py

```diff
@@ -0,0 +1,126 @@
+"""
+Support for lights under the iGlo brand.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/light.iglo/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.const import (CONF_HOST, CONF_NAME, CONF_PORT)
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS, ATTR_RGB_COLOR, ATTR_COLOR_TEMP,
+    SUPPORT_BRIGHTNESS, SUPPORT_COLOR_TEMP, SUPPORT_RGB_COLOR,
+    Light, PLATFORM_SCHEMA
+)
+
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['iglo==1.0.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'iGlo Light'
+DEFAULT_PORT = 8080
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the iGlo lighs."""
+    host = config.get(CONF_HOST)
+    name = config.get(CONF_NAME)
+    port = config.get(CONF_PORT)
+    add_devices([IGloLamp(name, host, port)])
+
+
+class IGloLamp(Light):
+    """Representation of an iGlo light."""
+
+    def __init__(self, name, host, port):
+        """Initialize the light."""
+        from iglo import Lamp
+        self._name = name
+        self._host = host
+        self._port = port
+        self._lamp = Lamp(0, host, port)
+        self.update()
+        self._on = True
+
+    @property
+    def name(self):
+        """Return the name of the light."""
+        return self._name
+
+    @property
+    def brightness(self):
+        """Return the brightness of this light between 0..255."""
+        return int((self._brightness / 200.0) * 255)
+
+    @property
+    def color_temp(self):
+        """Return the color temperature."""
+        return self._color_temp
+
+    @property
+    def min_mireds(self):
+        """Return the coldest color_temp that this light supports."""
+        return 1
+
+    @property
+    def max_mireds(self):
+        """Return the warmest color_temp that this light supports."""
+        return 255
+
+    @property
+    def rgb_color(self):
+        """Return the RGB value."""
+        return self._rgb
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return (SUPPORT_BRIGHTNESS | SUPPORT_COLOR_TEMP | SUPPORT_RGB_COLOR)
+
+    @property
+    def is_on(self):
+        """Return true if light is on."""
+        return self._on
+
+    def turn_on(self, **kwargs):
+        """Turn the light on."""
+        if not self._on:
+            self._lamp.switch(True)
+            self._on = True
```

Comment:
You shouldn't change any instance attributes that store state in the turn on/off methods. The `update` method will be called after `turn_on` or `turn_off` is called, since this is a polling entity, so state will be updated that way.